tests/python/pants_test/backend/python/tasks/test_select_interpreter.py

```diff
@@ -136,6 +136,31 @@ def test_interpreter_selection_invalidation(self):
     self.assertEqual('IronPython-2.99.999',
                       self._select_interpreter_and_get_version([tgtb], should_invalidate=False))
 
+  def test_invalidation_for_global_constraints(self):
+    # Because the system is setup with interpreter constraints, the task should
+    # invalidate on the first run.
+    self.assertEqual(
+      'IronPython-2.77.777',
+      self._select_interpreter_and_get_version([self.tgt1], should_invalidate=True)
+    )
+
+    self.set_options_for_scope(
+      PythonSetup.options_scope,
+      interpreter_constraints=RankedValue(RankedValue.CONFIG, ['IronPython>2.77.777'])
+    )
```

Comment:
This test currently fails due to the following assertion. 

I believe this is because this line is failing to properly update `PythonSetup`. Adding `print(PythonSetup.global_instance().interpreter_constraints)` right below this still prints the `'IronPython>=2.55'` used in this files `setUp()`.

Am I doing something wrong?